### DISCUSSION TITLE
Merge eclipse.platform.text into eclipse.platform.ui repository

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-platform/eclipse.platform.ui#433] [Ideas]
https://github.com/eclipse-platform/eclipse.platform.text could be merged to the platform repository, it is an important part of platform and evolving it here has many benefits:

1. No duplicate CI / releng handling required
2. better refactoring capabilities (e.g. changes spanning multiple repositories)
3. unified issue handling

has anyone concerns?

**Comment by iloveeclipse** [comment]
Yes.
1) In my experience platform text is more connected to platform ui and has less relationship to runtime/resources etc. So it simply doesn't fit.
2) What about build times?

**Reply by laeubi** [reply]
We already have some UI bundles here e.g.: https://github.com/eclipse-platform/eclipse.platform/tree/master/team/bundles so I'm not sure if there is a real separate UI<->non UI currently regarding to repositories but of course we can try to have some kind of at least trying it so in a glory future we end up with

1. https://github.com/eclipse-platform/eclipse.platform/
2. https://github.com/eclipse-platform/eclipse.platform.ui/

but this scattering of code really bugs me (and probably others as well).

about build times and it is a real concern one can build code independently even if its in the same repository but one better optimize the build/test rather than distributing the code all over many repositories.

**Comment by iloveeclipse** [comment]
It is not about "some UI bundles", it is about most important content being connected or not. platform UI and platform text are close connected, platform runtime/resources and platform text are not. It makes no sense to merge unrelated repositories.

**Reply by laeubi** [reply]
As said I would be fine with either... So I created:
- https://github.com/eclipse-platform/eclipse.platform.ui/issues/432

so we can transfer this discussion to platform.ui

**Comment by jukzi** [comment]
I agree with @iloveeclipse: please don't mix with runtime.
The main concern i have is that .ui has almost no active reviewing committers, while platform is almost PR free.

**Reply by laeubi** [reply]
@jukzi I moved this discussion now to the eclipse.ui (thanks @akurtakov for enabling discussions), and will adjust the topic now to prevent confusion

**Comment by mickaelistria** [comment]
Merging eclipse.platform.text into eclipse.platform.ui makes a lot of sense to me. The whole purpose of text framework is to serve code editors (even the non-UI parts are dedicated to serve the UI parts).

**Comment by vogella** [comment]
+1, even though this will increase the already very long build times of platform.ui.

**Comment by BeckerWdf** [comment]
In the meantime this has been done.
